src/journal-remote/microhttpd-util.c

```diff
@@ -72,20 +72,25 @@ int mhd_respond(struct MHD_Connection *connection,
                 enum MHD_RequestTerminationCode code,
                 const char *message) {
 
+        const char *fmt;
+
+        fmt = strjoina(message, "\n");
+
         return mhd_respond_internal(connection, code,
-                                    (char*) message, strlen(message),
+                                    fmt, strlen(message) + 1,
```

Comment:
I did that on purpose, to make it easier for the compiler to optimize. It's a five line function, so I think it's fine to do it this way.